applications/kserve/kserve/kserve-cluster-resources.yaml

```diff
@@ -649,3 +653,1383 @@ spec:
+        from environment\"\n          export NVSHMEM_IB_GID_INDEX=${NVSHMEM_IB_GID_INDEX:-$NCCL_IB_GID_INDEX}\n
+        \         export UCX_IB_GID_INDEX=${UCX_IB_GID_INDEX:-$NCCL_IB_GID_INDEX}\n
+        \         echo \"[Infer RoCE] Using hardcoded GID_INDEX=${NCCL_IB_GID_INDEX}
+        for NCCL, NVSHMEM, and UCX\"\n      elif [ -n \"$best_gid_index\" ]; then\n
+        \         echo \"[Infer RoCE] Selected GID_INDEX: ${best_gid_index} (found
+        on ${max_count} HCAs)\"\n\n          export NCCL_IB_GID_INDEX=${NCCL_IB_GID_INDEX:-$best_gid_index}\n
+        \         export NVSHMEM_IB_GID_INDEX=${NVSHMEM_IB_GID_INDEX:-$best_gid_index}\n
+        \         export UCX_IB_GID_INDEX=${UCX_IB_GID_INDEX:-$best_gid_index}\n\n
+        \         echo \"[Infer RoCE] Exported GID_INDEX=${best_gid_index} for NCCL,
+        NVSHMEM, and UCX\"\n      else\n          echo \"[Infer RoCE] ERROR: No valid
+        IPv4 ${KSERVE_INFER_IB_GID_INDEX_GREP} GID_INDEX found on any HCA.\"\n      fi\n
+        \ else\n      echo \"[Infer RoCE] No active HCAs found, skipping GID_INDEX
+        inference.\"\n  fi\nfi\n\nSTART_RANK=$(( ${LWS_WORKER_INDEX:-0} * {{ or .Spec.Parallelism.DataLocal
+        1 }} ))\neval \"vllm serve \\\n  /mnt/models \\\n  --served-model-name \"{{
+        .Spec.Model.Name }}\" \\\n  --port 8001 \\\n  --disable-log-requests \\\n
+        \ {{- if .Spec.Parallelism.Expert }}--enable-expert-parallel{{- end }} \\\n
+        \ {{- if .Spec.Parallelism.Tensor }}--tensor-parallel-size {{ .Spec.Parallelism.Tensor
+        }}{{- end }} \\\n  --data-parallel-size {{ or .Spec.Parallelism.Data 1 }}
+        \\\n  --data-parallel-size-local {{ or .Spec.Parallelism.DataLocal 1 }} \\\n
+        \ --data-parallel-address $(LWS_LEADER_ADDRESS) \\\n  --data-parallel-rpc-port
+        {{ if .Spec.Parallelism.DataRPCPort }}{{ .Spec.Parallelism.DataRPCPort }}{{
+        else }}5555{{- end }} \\\n  --data-parallel-start-rank $START_RANK \\\n  ${VLLM_ADDITIONAL_ARGS}
+        \\\n  --trust-remote-code \\\n  --headless\"\n  # BackendTLSPolicy is not
+        implemented yet so disable SSL for now\n  # --enable-ssl-refresh \\\n  # --ssl-certfile
+        \\\n  # /etc/ssl/certs/tls.crt \\\n  # --ssl-keyfile \\\n  # /etc/ssl/certs/tls.key\""
+      command:
+      - /bin/bash
+      - -c
+      env:
+      - name: HOME
+        value: /home
+      - name: VLLM_LOGGING_LEVEL
+        value: INFO
+      - name: HF_HUB_CACHE
+        value: /models
+      - name: VLLM_RANDOMIZE_DP_DUMMY_INPUTS
+        value: "1"
+      image: ghcr.io/llm-d/llm-d-dev:v0.2.2
+      imagePullPolicy: IfNotPresent
+      name: main
+      ports:
+      - containerPort: 8001
+        protocol: TCP
+      securityContext:
+        allowPrivilegeEscalation: false
+        capabilities:
+          add:
+          - IPC_LOCK
+          - SYS_RAWIO
+          - NET_RAW
+          drop:
+          - ALL
+        readOnlyRootFilesystem: false
+        runAsNonRoot: false
```

Comment:
We need to delete this insecure resource and make sure that only the kubeflow namespace is used. Maybe by adding namespace=kubeflow in the overlay.